python/test/test_binding.py

```diff
@@ -1030,12 +945,13 @@ def test_infer_async(self):
                 byte_size,
                 memory_type,
                 memory_type_id,
-                numpy_buffer,
             ) = out
+            ctypes_buffer = ctypes.create_string_buffer(byte_size)
+            ctypes.memmove(ctypes_buffer, out_buffer, byte_size)
+            numpy_buffer = numpy.frombuffer(ctypes_buffer, dtype=numpy.byte)
```

Comment:
who owns the memory in this case - 

does it become owned by numpy or ctypes_buffer ... just for my understanding

Reply:
The `out_buffer` is owned by the `res` object. The `ctypes_buffer` is a copy of the `out_buffer`, which the `ctypes_buffer` owns itself.

The `numpy_buffer` is not entirely clear from [the docs](https://numpy.org/doc/stable/reference/generated/numpy.frombuffer.html#numpy-frombuffer), but from my experiment it is referencing the `ctypes_buffer` as the `numpy_buffer` is marked as read-only and cannot interface with DLPack without a copy.